engine/swarm/secrets.md

```diff
@@ -0,0 +1,339 @@
+    container, substituting the container ID from the previous step.
+
+    ```bash
+    $ docker exec -it <container_id> cat /run/secrets/secret_data.txt
+
+    This is a secret
+    ```
+
+5.  Remove access to the secret from the running `redis` service.
+
+    ```bash
+    $ docker service update --secret-rm="secret_data.txt" redis
+    ```
+
+6.  Repeat steps 3 and 4 again, verifying that the service no longer has access
+    to the secret. The container ID will be different, because the
+    `service update` command redeploys the service.
+
+    ```bash
+    $ docker exec -it <container_id> cat /run/secrets/secret_data.txt
+
+    cat: can&apos;t open '/run/secrets/secret_data.txt': No such file or directory
+    ```
+
+7.  Stop and remove the service, and remove the secret from Docker.
+
+    ```bash
+    $ docker service rm redis
+
+    $ docker sercret rm secret_data.txt
+    ```
+
+## Example: Use secrets with a service
+
+In this example, you create a MySQL service with a custom username and password.
+Next, you add these credentials as secrets. Next, you create a WordPress
+service which uses these credentials to connect to MySQL. Finally, you update
+the MySQL password and update the secret, so that the WordPress service can
+still connect to MySQL.
+
+**Note**: This example uses a single-Engine swarm for simplicity.
+
+1.  On a new Docker Engine install, initialize a new swarm.
+    ```bash
+    $ docker swarm init
+    ```
+
+    This Engine is now the manager of a new swarm. If you join additional
+    managers or workers to this swarm, they will receive the key used to encrypt
+    and decrypt the secrets over a TLS-encrypted connection.
+
+2.  Create a new overlay network `wp_network` which the MySQL and WordPress
+    services will use to communicate.
+    ```bash
+    $ docker network create -d overlay wp_network
+    ```
+
+3.  Generate a random alphanumeric password for MySQL and store it as a Docker
+    secret with the name `mysql_password` using the `docker secret create`
+    command. To make the password shorter or longer, adjust the last argument of
+    the `openssl` command. This is just one way to create a relatively random
+    password. You can use another command to generate the password if you choose.
+    Notice that the password is never exposed on the command line or saved to a
+    plain-text file.
+
+    >**Note**: After you create a secret, you cannot update it. You can only
+    remove and re-create it, and you cannot remove a secret that a service is
+    using. However, you can grant or revoke a running service's access to
+    secrets using `service update`. If you need the ability to update a secret,
+    consider adding a version component to the secret name, so that you can later
+    add a new version, update the service to use it, then remove the old version.
+
+    ```bash
+    $ openssl rand -base64 20 |docker secret create mysql_password
+      lwxgks2rojdbvaui5kay1suvr
```

Comment:
Can you add something to the description like "after the `secret create` command completes, the ID of the secret is printed"

Looking at this example, I can mistake `lwxgks2rojdbvaui5kay1suvr` for being the actual secret